doc/source/cluster/kubernetes/user-guides/persist-kuberay-operator-logs.md

```diff
@@ -0,0 +1,110 @@
+
+# Persist KubeRay Operator Logs
+
+The KubeRay Operator plays a vital role in managing Ray clusters on Kubernetes. Persisting its logs is essential for effective troubleshooting and monitoring. This guide describes methods to set up centralized logging for KubeRay Operator logs.
+
+## Grafana Loki
+
+[Grafana Loki][GrafanaLoki] is a log aggregation system optimized for Kubernetes, providing efficient log storage and querying. The following steps set up [Fluent Bit][FluentBit] as a DaemonSet to collect logs from Kubernetes containers and send them to Loki for centralized storage and analysis.
+
+### Deploy Loki monolithic mode
+
+Loki’s Helm chart supports three deployment methods to fit different scalability and performance needs: Monolithic, Simple Scalable, and Microservices. Here, we will demonstrate the monolithic method. For details on each deployment mode, see the [Loki deployment](https://grafana.com/docs/loki/latest/get-started/deployment-modes/) modes documentation.
+
+Deploy the Loki deployment with the [Helm chart repository](https://github.com/grafana/loki/tree/main/production/helm/loki).
+
+```shell
+helm repo add grafana https://grafana.github.io/helm-charts
+helm repo update
+
+# Install Loki with single replica mode
+helm install loki grafana/loki -f https://raw.githubusercontent.com/grafana/loki/refs/heads/main/production/helm/loki/single-binary-values.yaml
+```
+
+### Configure log processing
+
+Create a `fluent-bit-config.yaml` file which configures Fluent Bit to:
+
+* Tail log files from Kubernetes containers.
+* Parse multi-line logs for Docker and CRI (Container Runtime Interface) formats.
+* Enrich logs with Kubernetes metadata such as namespace, pod, and container names.
+* Send the logs to Loki for centralized storage and querying.
+```{literalinclude} ../configs/loki.log.yaml
+:language: yaml
+:start-after: Fluent Bit Config
+:end-before: ---
+```
+
+A few notes on the above config:
+
+* Inputs: The `tail` input reads log files from `/var/log/containers/*.log`, with `multiline.parser` to handle complex log messages across multiple lines.
+* Filters: The `kubernetes` filter adds metadata like namespace, pod, and container names to each log, enabling more efficient log management and querying in Loki.
+* Outputs: The `loki` output block specifies Loki as the target. The `Host` and `Port` define the Loki service endpoint, and `Labels` add metadata for easier querying in Grafana. Additionally, `tenant_id` allows for multi-tenancy if required by the Loki setup.
+
+Deploy the Fluent Bit deployment with the [Helm chart repository](https://github.com/fluent/helm-charts/tree/main/charts/fluent-bit).
+
+```shell
+helm repo add fluent https://fluent.github.io/helm-charts
+helm repo update
+
+helm install fluent-bit fluent/fluent-bit -f fluent-bit-config.yaml
+```
+
+### Install the KubeRay Operator and Deploy a RayCluster
+
+Follow [this document](kuberay-operator-deploy) to install the KubeRay operator and deploy a RayCluster.
```

Comment:
"kuberay-operator-deploy" points to step 2 in the doc. Can we also add a reference to step 3, which provides instructions for deploying a RayCluster?

Reply:
sure!